libfreerdp/common/assistance.c

```diff
@@ -184,7 +184,7 @@ int freerdp_assistance_parse_address_list(rdpAssistanceFile* file, char* list)
 		file->MachinePorts[i] = (UINT32) atoi(q);
 
 		if (!file->MachineAddresses[i])
-			return -1;
+			goto out;
```

Comment:
good catch.
We should also clean up the rest (file->MachineAddresses[i]) if a strdup fails maybe you can add that as well.